### PR TITLE
STIJ-19: Added themify mixin and link + headings example.

### PR DESCRIPTION
**Pull request by meirege**

### CODE REVIEW

File: components/01-mixins/themify/_themify.scss [hunk 1]
```diff
@@ -0,0 +1,47 @@
+//
+// Themify mixin
+// @param {Map} $themes [$themes] - Map of themes to use
+//
+@mixin themify($themes: $themes) {
+  @each $theme, $colors in $themes {
+
+    &,
```
Comment (daften) [inline components/01-mixins/themify/_themify.scss]: Using the element itself in this themify function, won't this generate 5 h1 declarations where the last one "wins"? What's the point in that?

File: components/21-atoms/headings/heading-1/_heading-1.scss [hunk 2]
```diff
@@ -1,23 +1,25 @@
...
-  margin-bottom: 1.5rem;
-  color: $heading-1-color;
-  font-family: $heading-1-font-family;
-  font-size: 1.6rem;
-  font-weight: 700;
-  line-height: 1.1em;
-
-  a {
+  @include themify {
+    margin-bottom: 1.5rem;
```
Comment (daften) [inline components/21-atoms/headings/heading-1/_heading-1.scss]: The themify function is mainly to differentiate color schemes in our case. Do we want these general layouting items included here? I'd say that will generate more code than necessary.

File: gulpfile.js [hunk 3]
```diff
@@ -52,7 +52,7 @@ var imagemin = require('gulp-imagemin');
 var pngquant = require('imagemin-pngquant');
 var es = require('event-stream');
 var minify = require('gulp-minify');
-var csscomb =require('gulp-csscomb');
+//var csscomb =require('gulp-csscomb');
```
Comment (daften) [inline gulpfile.js]: How is this related?

File: components/00-settings/_vars.scss [hunk 4]
```diff
@@ -168,3 +157,179 @@ $cta-link-icon-background-color: $color-blue !default;
 $cta-link-icon-color: $color-white !default;
 
 $cta-link-title-color: $color-darkest-gray !default;
+
+//
+// THEMIFY SETTINGS.
+//
+$themes: (
```
Comment (daften) [inline components/00-settings/_vars.scss]: While this is a big map, it makes it clear what variables are used per section IMO. What we would need is documentation about each variable, what it will/should do.

Comment (daften) [inline components/00-settings/_vars.scss]: We also need a strategy on what we will do when variables here are added, how anything using this styleguide should adapt to that. Changelog is one thing we'll need to maintain carefully, update guides between versions?
Merging of maps was not an option if I remember correctly from @meirege . We should discuss this further ASAP

File: components/00-settings/_vars.scss [hunk 5]
```diff
@@ -77,18 +77,7 @@ $bg-hover-breadcrumb: $color-lightest-blue !default;
...
-// Link settings.
-$link-color: $color-blue !default;
-$link-color-hover: $color-dark-blue !default;
-$link-color-visited: $color-dark-blue !default;
-$link-color-active: $color-dark-blue !default;
-$link-color-focus: $color-dark-blue !default;
-$link-color-in-button: $color-white !default;
-
 // Heading 1 settings.
-$heading-1-color: $color-darkest-blue !default;
```
Comment (daften) [inline components/00-settings/_vars.scss]: Shouldn't heading2 and so on colors be removed too?

File: components/21-atoms/link/_link.scss [hunk 6]
```diff
@@ -1,23 +1,32 @@
...
-  color: $link-color;
+  @include themify {
+    color: $link-color;
+  }
+
+  text-decoration: none;
 
   &:visited {
-    color: $link-color-visited;
+    @include themify {
```
Comment (daften) [inline components/21-atoms/link/_link.scss]: why separate themify's and not include it in the main one?

File: gulpfile.js (hunk 3 above)
Comment (Jeroen005) [reply gulpfile.js]: CssComb doesn't work with complex sass syntax like used for the themify mixins.

File: components/01-mixins/general/_mixins.scss [hunk 7]
```diff
@@ -4,14 +4,6 @@
 //
 
 //
-// Border-box polyfill helper.
-// @include border-box;
-//
-@mixin border-box() {
-  box-sizing: border-box;
-}
```
Comment (meirege) [inline components/01-mixins/general/_mixins.scss]: Why delete this? Did you add it somewhere as a default perhaps?

File: components/11-base/base/_base.scss [hunk 8]
```diff
@@ -11,10 +11,10 @@
 }
 
 html {
-  @include border-box;
   background: $bg-outside-container;
   font: $default-font-weight $default-font-size $default-font-family;
   line-height: $default-line-height;
+  box-sizing: border-box;
```
Comment (meirege) [inline components/11-base/base/_base.scss]: Should't this be on * instead of only html?

File: components/21-atoms/inputs/input-checkbox/_input-checkbox.scss [hunk 9]
```diff
@@ -1,4 +1,6 @@
 input[type="checkbox"] + span {
-  color: $color-dark-gray;
+  @include themify {
+    color: $label-color;
+  }
```
Comment (meirege) [inline components/21-atoms/inputs/input-checkbox/_input-checkbox.scss]: I think the validator will complain. Isn't a whitespace necessary here?

Comment (Jeroen005) [reply components/21-atoms/inputs/input-checkbox/_input-checkbox.scss]: It isn't necessary for the validator.

File: components/11-base/base/_base.scss (hunk 8 above)
Comment (Jeroen005) [reply components/11-base/base/_base.scss]: * is inherit so we can overwrite it a bit easier. https://css-tricks.com/inheriting-box-sizing-probably-slightly-better-best-practice/

File: components/01-mixins/general/_mixins.scss (hunk 7 above)
Comment (Jeroen005) [reply components/01-mixins/general/_mixins.scss]: In the old theme there where vendor prefixes in this mixin. since we use autoprefixer there is no need anymore for this mixin.
